src/AWSMetadataUtilities.jl

```diff
@@ -0,0 +1,374 @@
+        operation = operation[2]
+        name = operation["name"]
+        method = operation["http"]["method"]
+        request_uri = operation["http"]["requestUri"]
+
+        documentation = ""
+
+        if haskey(operation, "documentation")
+            documentation = _documentation_cleaning(operation["documentation"])
+        end
+
+        required_parameters = Dict()
+        optional_parameters = Dict()
+
+        if haskey(operation, "input")
+            required_parameters, optional_parameters = _get_function_parameters(operation["input"]["shape"], shapes)
+        end
+
+        operation_definition = _generate_high_level_definition(
+            service_name,
+            protocol,
+            name,
+            method,
+            request_uri,
+            required_parameters,
+            optional_parameters,
+            documentation
+        )
+
+        push!(operation_definitions, operation_definition)
+    end
+
+    println(typeof(operation_definitions))
+
+    return operation_definitions
+end
+
+"""
+    _generate_high_level_definition(
+        service_name::String,
+        protocol::String,
+        name::String,
+        method::String,
+        request_uri::String,
+        required_parameters::Dict,
+        optional_parameters::Dict,
+        documentation::String
+    )
+
+Generate the high-level definition for a services function.
+
+# Arguments
+- `name::String`: Name of this services function
+- `service_name::String`: Name of the AWS Service
+- `protocol::String`: Request protocol for the AWS Service
+- `method::String`: Request method for this services function
+- `request_uri::String`: Endpoint for this services function
+- `required_parameters::Dict{Any, Any}`: Required parameters for this services function
+- `optional_parameters::Dict{Any, Any}`: Optional parameters for this services function
+- `documentation::String`: Documentation for this services function
+
+# Return
+- `String`: High-level definition in Julia code for the services function
+"""
+function _generate_high_level_definition(
+    service_name::String,
+    protocol::String,
+    name::String,
+    method::String,
+    request_uri::String,
+    required_parameters::Dict,
+    optional_parameters::Dict,
+    documentation::String
+)
+    required_param_keys = collect(keys(required_parameters))
+
+    operation_definition = """
+    \"\"\"
+        $name()
+
+    $documentation
+    """
+
+    # Add in the required parameters if applicable
+    if !isempty(required_parameters)
+        operation_definition = operation_definition * """
+
+        Required Parameters
+        $(json(required_parameters, 2))"""
+    end
+
+    # Add in the optional parameters if applicable
+    if !isempty(optional_parameters)
+        operation_definition = operation_definition * """
+
+        Optional Parameters
+        $(json(optional_parameters, 2))"""
+    end
+
+    operation_definition = operation_definition * "\"\"\""
```

Comment:
```suggestion
    operation_definition *= "\"\"\""
```
Maybe easier to read `repeat('"', 3)` instead of `"\"\"\""`